skltemplate/template.py

```diff
@@ -47,4 +51,135 @@ def predict(self, X):
             Returns :math:`x^2` where :math:`x` is the first column of `X`.
         """
         X = check_array(X)
-        return X[:, 0]**2
\ No newline at end of file
+        return X[:, 0]**2
+
+
+class TemplateClassifier(BaseEstimator, ClassifierMixin):
+    """ An example classifier which implements a 1-NN algorithm.
+
+    Parameters
+    ----------
+    demo_param : str, optional
+        A parameter used for demonstation of how to pass and store paramters.
+
+    Attributes
+    ----------
+    X_ : array, shape = [n_samples, n_features]
+        The input passed during :meth:`fit`
+    y_ : array, shape = [n_samples]
+        The labels passed during :meth:`fit`
+    """
+    def __init__(self, demo_param='demo'):
+        self.demo_param = demo_param
+
+    def fit(self, X, y):
+        """A reference implementation of a fitting function for a classifier.
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+            The training input samples.
+        y : array-like, shape = [n_samples]
+            The target values. An array of int.
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        # Check that X and y have correct shape
+        X, y = check_X_y(X, y)
+        # Store the classes seen durind fit
+        self.classes_ = unique_labels(y)
```

Comment:
Using `check_classification_targets` from utils.multiclass should be a better way of testing this, especially since it was made for this specific purpose.